packages/aws-cdk-lib/aws-ecs/test/cluster.test.ts

```diff
@@ -2483,6 +2524,23 @@ describe('cluster', () => {
 
   });
 
+  test('should throw an error if the capacity provider was created using an imported AsgCapacityProvider', () => {
+    // GIVEN
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app, 'test');
+    const autoScalingGroup = autoscaling.AutoScalingGroup.fromAutoScalingGroupName(stack, 'ASG', 'my-asg');
+    const cluster = new ecs.Cluster(stack, 'EcsCluster');
+
+    const capacityProvider = new ecs.AsgCapacityProvider(stack, 'provider', {
+      autoScalingGroup,
+      enableManagedTerminationProtection: false,
```

Comment:
```suggestion
      enableManagedTerminationProtection: true,
```

I guess this should be true?

Reply:
No, this test is for the 2nd case from the PR description:

> It will also throw a clear exception when calling Cluster.addAsgCapacityProvider with an AsgCapacityProvider created with an imported ASG.

Reply:
OK I got you.

If your assertion is

>It will also throw a clear exception when calling Cluster.addAsgCapacityProvider with an AsgCapacityProvider created with an imported ASG.

How about change

```ts
 test('should throw an error if the capacity provider was created using an imported AsgCapacityProvider', () => {
```
to
 
 ```ts
  test('throws when calling Cluster.addAsgCapacityProvider with an AsgCapacityProvider created with an imported ASG', () => {
```
 (because it's the ASG that was imported, not capacity provider)
 
And I would assign that ASG to `importedAsg` as below. This might make it more clear?

```ts
// GIVEN
    const app = new cdk.App();
    const stack = new cdk.Stack(app, 'test');
    const importedAsg = autoscaling.AutoScalingGroup.fromAutoScalingGroupName(stack, 'ASG', 'my-asg');
    const cluster = new ecs.Cluster(stack, 'EcsCluster');

    const cp = new ecs.AsgCapacityProvider(stack, 'provider', {
                           autoScalingGroup: importedAsg
                           enableManagedTerminationProtection: false,
                        });

```